File: 07-Builder/src/go/director.go
package main

type HouseDirector struct{}

func (d HouseDirector) BuildLuxuryHouse() House {
	return NewHouseBuilder().
		Rooms(6).Floors(3).
		Garage().Pool().Garden().
		RoofType("slate").
		Build()
}

func (d HouseDirector) BuildSimpleHouse() House {
	return NewHouseBuilder().
		Rooms(2).Floors(1).
		RoofType("shingle").
		Build()
}

func (d HouseDirector) BuildFamilyHouse() House {
	return NewHouseBuilder().
		Rooms(4).Floors(2).
		Garage().Garden().
		RoofType("tile").
		Build()
}

File: 07-Builder/src/rust/director.rs
use crate::house::{House, HouseBuilder};

pub struct HouseDirector;

impl HouseDirector {
    pub fn build_luxury_house() -> House {
        HouseBuilder::new()
            .rooms(6)
            .floors(3)
            .garage()
            .pool()
            .garden()
            .roof_type("slate")
            .build()
    }

    pub fn build_simple_house() -> House {
        HouseBuilder::new()
            .rooms(2)
            .floors(1)
            .roof_type("shingle")
            .build()
    }

    pub fn build_family_house() -> House {
        HouseBuilder::new()
            .rooms(4)
            .floors(2)
            .garage()
            .garden()
            .roof_type("tile")
            .build()
    }
}
